libraries/fullscreenintent/api/src/main/kotlin/io/element/android/libraries/fullscreenintent/api/FullScreenIntentPermissionsStateProvider.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package io.element.android.libraries.fullscreenintent.api
+
+fun aFullScreenIntentPermissionsState(
```

Comment:
Yes, I hesitated, but all our `StateProvider` files contains a function which create state class instances (for instance `aMessageComposerState`) and those function are in production code, but I admit, generally in `impl` modules.

So that's maybe not a big deal?

Reply:
For the name, I keep the name we usually have for state provider... That does not exist, but I could create one, which could be used here: https://github.com/element-hq/element-x-android/blob/241dae14f44bb97281531634ac1d627d46e4b0f8/features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/components/FullScreenIntentPermissionBanner.kt#L31

But since the View is only using the lambda, I guess it does not worth it?